src/codegen/generate.hpp

```diff
@@ -58,7 +57,7 @@ class NasmGenerator_x86_64
   }
 
   void emitGlobalDirective(const std::string_view &name) {
-    textSection.globalDeclarations << INDENT << "global " << name << ENDL;
+    textSection.globalDeclarations << INDENT << "global " << name << ":function" << ENDL;
```

Comment:
Neither me or chat gpt get why we need the `:function` here. It does not work without it and we had it in the ass version of the stdlib so I added it. Maybe you are aware of the reason why we need this modifier

Reply:
This is just required in nasm x86 flavour syntax, it's necessary to generate the symbol table in the object file later on

Reply:
I don't see any reason not to add it every time we expose a function, so the way you did it is perfect